tests/test_tem.py

```diff
@@ -1,42 +1,248 @@
+    )
+    tem_simulator.output_path_dict["log_file"] = str(
+        Path(cwd, tmp_path, out_file_name + ".log")
+    )
+    tem_simulator.output_path_dict["inp_file"] = str(
+        Path(cwd, tmp_path, out_file_name + ".inp")
+    )
+    tem_simulator.output_path_dict["h5_file"] = str(
+        Path(cwd, tmp_path, out_file_name + ".h5")
+    )
+    tem_simulator.output_path_dict["h5_file_noisy"] = str(
+        Path(cwd, tmp_path, out_file_name + "-noisy.h5")
+    )
+    tem_simulator.output_path_dict["pdb_file"] = str(
+        Path(cwd, test_files_path, "4v6x.pdb")
+    )
+
+    return tem_simulator
+
+
+@pytest.fixture
+def sample_resources():
+    """Return sample resources for testing."""
+    test_files_path = "./tests/test_files"
+    cwd = os.getcwd()
+    resources = {}
+
+    resources["files"] = {
+        "path_yaml": str(Path(cwd, test_files_path, "path_config.yaml")),
+        "sim_yaml": str(Path(cwd, test_files_path, "sim_config.yaml")),
+        "pdb_file": str(Path(cwd, test_files_path, "4v6x.pdb")),
+    }
+
+    micrograph = np.load(str(Path(cwd, test_files_path, "micrograph.npz")))
+
+    resources["data"] = {"micrograph": micrograph.f.arr_0}
+
+    return resources
+
+
+def test_temsimulator_constructor(sample_resources):
+    """Test whether constructor populates attributes."""
+    tem_sim = tem.TEMSimulator(
+        sample_resources["files"]["path_yaml"], sample_resources["files"]["sim_yaml"]
+    )
+    parameters_dict_keys = [
+        "simulation",
+        "sample",
+        "particle",
+        "particleset",
+        "beam",
+        "optics",
+        "detector",
+    ]
+    assert tem_sim.output_path_dict is not None
+    assert tem_sim.sim_dict is not None
+    assert set(parameters_dict_keys).issubset(tem_sim.parameter_dict.keys())
+
+
+def test_get_config_from_yaml(sample_resources, sample_class):
+    """Test whether yaml is parsed."""
+    expected_config_template = {
+        "beam_parameters": 4,
+        "optics_parameters": 10,
+        "detector_parameters": 11,
+        "specimen_grid_params": 3,
+        "molecular_model": 3,
+    }
+
+    test_yaml = sample_resources["files"]["sim_yaml"]
+    returned_config = sample_class.get_config_from_yaml(test_yaml)
+
+    for config_group, config_list in returned_config.items():
+        assert config_group in expected_config_template
+        assert len(config_list) is expected_config_template[config_group]
+
+
+def test_classify_input_config(sample_class):
+    """Test classification of simulation parameters."""
+    raw_params = {
+        "molecular_model": {
+            "voxel_size_nm": 0.1,
+            "particle_name": "toto",
+            "particle_mrcout": "None",
+        },
+        "specimen_grid_params": {
+            "hole_diameter_nm": 1200,
+            "hole_thickness_center_nm": 100,
+            "hole_thickness_edge_nm": 100,
+        },
+        "beam_parameters": {
+            "voltage_kV": 300,
```

Comment:
does `voltage_kV` have to match the non capitalization in https://github.com/compSPI/simSPI/blob/tem_with_tests/tests/test_files/sim_config.yaml#L12

also there might be a mismatch in `energy_spread_V`

maybe also change `detector_Nx_px`, `detector_Nx_py`, `detector_Q_efficiency`, `MTF_params` to have no caps

Reply:
Changes made as suggested - the changes to detector_nx_px and co. are for the best (the capitals were a carryover from imported code) but they also required changes to cryoemio in ioSPI. A new branch "parameter_name_changes" has been opened in ioSPI to this end, so please take a look at https://github.com/compSPI/ioSPI/pull/27 so that these changes can be imported.

Reply:
Note: this is also why the tests are failing.